cadl-extension/src/code-model-builder.ts

```diff
@@ -624,6 +625,56 @@ export class CodeModelBuilder {
     if (op.convenienceApi) {
       this.trackSchemaUsage(schema, { usage: [SchemaContext.ConvenienceApi] });
     }
+
+    if (!schema.language.default.name && schema instanceof ObjectSchema) {
+      // anonymous model
+
+      if (!parameter.language.default.name) {
+        // name the parameter for documentation
+        parameter.language.default.name = "request";
+      }
+      this.trackSchemaUsage(schema, { usage: [SchemaContext.Anonymous] });
+
+      if (op.convenienceApi && op.parameters) {
+        op.convenienceApi.parameters = [];
+        for (const [key, _] of parameters.properties) {
+          const existParameter = op.parameters.find((it) => it.language.default.serializedName === key);
+          if (existParameter) {
+            // parameter
+            if (
+              existParameter.implementation === ImplementationLocation.Method &&
+              (existParameter.origin?.startsWith("modelerfour:synthesized/") ?? true)
+            ) {
+              op.convenienceApi.parameters.push(existParameter);
+            }
+          } else {
+            // property from anonymous model
+            const existBodyProperty = schema.properties?.find((it) => it.serializedName === key);
+            if (existBodyProperty) {
+              op.convenienceApi.parameters.push(
+                new VirtualParameter(
+                  existBodyProperty.language.default.name,
+                  existBodyProperty.language.default.description,
+                  existBodyProperty.schema,
+                  {
+                    originalParameter: parameter,
+                    targetProperty: existBodyProperty,
+                    language: {
+                      default: {
+                        serializedName: existBodyProperty.serializedName,
+                      },
+                    },
+                  },
+                ),
+              );
+            }
+          }
+        }
+      }
+
+      // TODO remove the line after full support on anonymous model
+      op.convenienceApi = undefined;
```

Comment:
convenience API disable before full support.

protocol API is good.

Reply:
ok, so the alias name will not be used? in our case, the name `Request`.
```
 alias Request = { 
   user?: User; 
   input: string; 
 }; 
```

Reply:
alias is only exist when writing Cadl source.

It is not visible from emitter.